packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/Reflow/DynamicData.jsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import Button from 'terra-button/lib/Button';
+import utils from '@cerner/carbon-graphs/lib/js/helpers/utils';
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '@cerner/terra-graphs-docs/lib/Css/ExampleGraphContainer.module.scss';
+import '@cerner/terra-graphs/lib/components/Graph.module.scss';
+import '@cerner/terra-graphs/lib/components/Timeline/Timeline.module.scss';
+import data1 from '@cerner/terra-graphs-docs/src/example-datasets/dataObjects/Timeline/defaultDataset';
+import data2 from '@cerner/terra-graphs-docs/src/example-datasets/dataObjects/Timeline/criticalDataset';
+import getTimelinePanningConfig from '@cerner/terra-graphs-docs/lib/example-datasets/graphConfigObjects/Timeline/timelinePanningConfig';
+
+/*
+Please refer to the documentation below to see the graphConfig and data objects
+*/
+
+const graphConfig = utils.deepClone(getTimelinePanningConfig('#TimelinePanningDynamicExample'));
+const dataset = [utils.deepClone(data1)];
+
+const initialState = {
+  initial: 0,
+  factor: 3,
+};
+
+let graph;
+
+const TimelinePanningDynamicDataExample = () => {
+  React.useEffect(() => {
+    graph = Carbon.api.timeline(graphConfig);
+    dataset.forEach((data) => {
+      graph.loadContent(data);
+    });
+  }, []);
+
+  const reducer = (panState, action) => {
+    let hour;
+
+    switch (action.type) {
+      case 'panLeft':
+        hour = panState.initial - panState.factor;
+        break;
+      case 'panRight':
+        hour = panState.initial + panState.factor;
+        break;
+      default:
+        return panState;
+    }
+
+    graph.config.axis.x.lowerLimit = new Date(2016, 0, 1, hour).toISOString();
```

Comment:
Typically, your reducer shouldn't have any side effects. The reducer should determine and return the new state; you could add a separate effect for other changes.

```jsx
const [state, dispatch] = React.useReducer(reducer, initialState);

React.useLayoutEffect(() => {
  if (!graph) { return; }
  
  graph.config.axis.x.lowerLimit = new Date(2016, 0, 1, state.initial).toISOString();
  graph.config.axis.x.upperLimit = new Date(2016, 0, 2, state.initial).toISOString();
  
  const newDataset = {
    panData: utils.deepClone(data2),
  };

  graph.reflowMultipleDatasets(newDataset);
}, [graph, state.initial]);
```

Reply:
Fixed [here](https://github.com/cerner/terra-graphs/pull/193/commits/4f472d55aa757d4dd6e033b6e1c90aa0dcc4095a).